src/Cake.Prca/Issues/CodeAnalysisIssue.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace Cake.Prca.Issues
+{
+    using System;
+    using Core.IO;
+
+    /// <summary>
+    /// Base class for a code analysis issue.
+    /// </summary>
+    public class CodeAnalysisIssue : ICodeAnalysisIssue
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CodeAnalysisIssue"/> class.
+        /// </summary>
+        /// <param name="filePath">The path to the file affacted by the issue.
+        /// The path needs to be relative to the repository root.</param>
+        /// <param name="line">The line in the file where the issues has occurred.</param>
+        /// <param name="message">The message of the code analysis issue.</param>
+        /// <param name="priority">The priority of the message used to filter out issues if there are more issues than
+        /// should be posted.</param>
+        /// <param name="rule">The rule of the code analysis issue.</param>
+        public CodeAnalysisIssue(string filePath, int line, string message, int priority, string rule)
+        {
+            filePath.NotNullOrWhiteSpace(nameof(filePath));
+            line.NotNegativeOrZero(nameof(line));
```

Comment:
I think this constraint is too strict. If I remember correctly I've seen cases especially for FxCop where the line number couldn't be correctly determined in the rule implementation and is 0 or even -1. Can't remember only one of both of them. I've also seen roslyn rules producing such line number, e.g. if you have a rule which applies to the file as a whole like [SA1412](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1412.md).